.github/workflows/config/spellcheck_config.yml

```diff
@@ -0,0 +1,144 @@
+      - alt
+      ignores:
+      - ':matches(code, pre)'
+      - 'code'
+      - 'pre'
+
+- name: rst
+  sources:
+  - '**/*.rst'
+  expect_match: false
+  aspell:
+    lang: en
+    d: en_US
+  dictionary:
+    wordlists:
+    - .github/workflows/config/spelling_allowlist.txt
+  pipeline:
+  - pyspelling.filters.markdown:
+      markdown_extensions:
+      - markdown.extensions.extra:
+  - pyspelling.filters.html:
+      comments: false
+      attributes:
+      - title
+      - alt
+      ignores:
+      - ':matches(code, pre)'
+      - 'code'
+      - 'pre'
+  - pyspelling.filters.context:
+      context_visible_first: true
+      delimiters:
+      # Ignore lines that start with two dots
+      - open: '^\s*\.\.'
+        close: '$'
+
+- name: html
+  sources:
+  - '**/*.html'
+  expect_match: true
+  aspell:
+    lang: en
+    d: en_US
+  dictionary:
+    wordlists:
+    - .github/workflows/config/spelling_allowlist.txt
+  pipeline:
+  - pyspelling.filters.html:
+      attributes:
+      - title
+      - alt
+      ignores:
+      - code
+      - pre
+
+- name: cxx
+  sources:
+  - '**/*.h'
+  - 'docs/sphinx/examples/**/*.cpp'
+  expect_match: false
+  aspell:
+    lang: en
+    d: en_US
+  dictionary:
+    wordlists:
+    - .github/workflows/config/spelling_allowlist.txt
+  pipeline:
+  - pyspelling.filters.url:
+  - pyspelling.filters.context:
+      context_visible_first: true
+      escapes: '\\[\\`~]'
+      delimiters:
+      # Ignore comments of the form `// namespace ...` since they are used to annotate the closing namespace
+      - open: '\/{2}\s*namespace\s+'
+        close: '($|\s)'
+      # Ignore comments of the form `/*...*/` since they are used to annotate argument names
+      - open: '\/\*'
+        close: '\*\/'
+        content: \S*?        
+  - pyspelling.filters.cpp:
+      block_comments: true
+      line_comments: true
+      group_comments: true
+      strings: false
+  - pyspelling.filters.context:
+      context_visible_first: true
+      escapes: '\\[\\`~]'
+      delimiters:
+      # Ignore multiline content between three or more backticks
+      - open: '(?s)^(?P<open>\s*`{3,})\S*\s*$'
+        close: '^(?P=open)$'
+      # Ignore content between inline backticks
+      - open: '(?P<open>`+)'
+        close: '(?P=open)'
+
+- name: python
+  sources:
+  - '**/*.py'
+  expect_match: false
+  aspell:
+    lang: en
+    d: en_US
+  dictionary:
+    wordlists:
+    - .github/workflows/config/spelling_allowlist.txt
+  pipeline:
+  - pyspelling.filters.url:
+  - pyspelling.filters.python:
+      docstrings: true
+      comments: true
+      strings: true
+  - pyspelling.filters.context:
+      context_visible_first: true
+      escapes: '\\[\\`~]'
+      delimiters:
+      # Ignore multiline content between three or more backticks
+      - open: '(?s)^(?P<open>\s*`{3,})\S*\s*$'
+        close: '^(?P=open)$'
+      # Ignore content between inline backticks
+      - open: '(?P<open>`+)'
+        close: '(?P=open)'
```

Comment:
So the backticks aren't markdown necessarily, but a way to defeat the spellchecker. What about `<code>` .. `</code>` blocks?

Also, can we just filter out any word that is written in camelCase or CamelCase without requiring backticks?

Reply:
I think it is helpful if we can distinguish somehow what is code from what is "normal language". To make the spellchecker happy we could regex match any words that contain a capital letter that is not at the beginning of the word and ignore them. We can treat all matches as code and can at some point in the future also validate that documentation for this parameter/class name/whatever else actually exists. The reason that I am a bit less fond of automatic detection of camel casing is that you could always have a simple class name `foo` that is not distinguishable by looking for camel case. I hence think there necessarily always needs to be a way to explicitly mark words as code, and we should just use that everywhere. Whether that's backticks or xml doesn't matter to me, but we need to make sure doxygen or sphinx is properly recognizing it as code.

Reply:
OK. Yeah, it's a catch-22 sort of situation.

Reply:
Doxygen seems to be rather good at auto detecting references to code without any kind of marker. I hence added a filter for the spellcheck to ignore words that contains capital letters, underscores, or numbers. However, other things, such as e.g. when referring to a parameter name `idx` would need to be fenced with backticks.